cluster/terraform_kubernetes/grafana.tf

```diff
@@ -0,0 +1,119 @@
+data "azurerm_key_vault" "key_vault_grafana" {
+  name                = var.cluster_kv
+  resource_group_name = var.resource_group_name
+}
+
+data "azurerm_key_vault_secret" "grafana_admin_user" {
+  name         = "GRAFANA-ADMIN-USER"
+  key_vault_id = data.azurerm_key_vault.key_vault_grafana.id
+}
+
+data "azurerm_key_vault_secret" "grafana_admin_password" {
+  name         = "GRAFANA-ADMIN-PASSWORD"
+  key_vault_id = data.azurerm_key_vault.key_vault_grafana.id
+}
+
+
+resource "kubernetes_config_map" "grafana_datasources" {
+  metadata {
+    name      = "grafana-datasources"
+    namespace = "monitoring"
```

Comment:
you'll have to make this` namespace = kubernetes_namespace.default_list["monitoring"].metadata[0].name` for all resources, otherwise cluster build will fail sometimes on the namespace not being available